internal/e2e/tester/game/main.go

```diff
@@ -14,41 +17,64 @@ import (
 )
 
 func main() {
-	namespace := os.Getenv("NAMESPACE")
 	options := []cardinal.WorldOption{
-		cardinal.WithNamespace(namespace),
 		cardinal.WithReceiptHistorySize(10), //nolint:gomnd // fine for testing.
 	}
-	if os.Getenv("ENABLE_ADAPTER") == "false" {
+	// if os.Getenv("ENABLE_ADAPTER") == "false" {
+	if true { // uncomment above to enable adapter from env.
 		log.Println("Skipping adapter")
 	} else {
 		options = append(options, cardinal.WithAdapter(setupAdapter()))
 	}
-
-	world, err := cardinal.NewWorld(os.Getenv("REDIS_ADDR"), "", options...)
+	world, err := cardinal.NewWorld(options...)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatal(err, eris.ToString(err, true))
 	}
 	err = errors.Join(
 		cardinal.RegisterComponent[comp.Location](world),
 		cardinal.RegisterComponent[comp.Player](world),
 	)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatal(err, eris.ToString(err, true))
 	}
 	err = cardinal.RegisterMessages(world, msg.JoinMsg, msg.MoveMsg)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatal(err, eris.ToString(err, true))
+	}
+
+	err = cardinal.RegisterQueryWithEVMSupport[query.LocationRequest, query.LocationReply](
+		world,
+		"location",
+		func(ctx cardinal.WorldContext, req *query.LocationRequest) (*query.LocationReply, error) {
```

Comment:
could also just have this be an exported function from `query/location.go`

Reply:
That would be just a function that takes the world. Sure. I don't mind this.

Reply:
should just be the same function, no? they get access to WorldCtx

Reply:
oh nvm i see how you did it, that works for me 🤙🏻

Reply:
RegisterQueryWithEVMSupport can only be called from main because it needs to be in the context of world. World is the only reason why I moved everything here.

Reply:
yep yep, this way works, i originally meant to just export the _handler_ logic (the query fn itself) and then pass that in. 

but this way also works!